app/lib/features/events/pages/event_page.dart

```diff
@@ -167,84 +193,194 @@ class CalendarEventPage extends ConsumerWidget {
+                                  horizontal: 15,
+                                  vertical: 5,
                                 ),
-                                child: Text(
-                                  'Join',
-                                  style: Theme.of(context).textTheme.bodyMedium,
+                                child: Row(
+                                  crossAxisAlignment: CrossAxisAlignment.start,
+                                  children: [
+                                    Text(
+                                      'Description: ',
+                                      style: Theme.of(context)
+                                          .textTheme
+                                          .bodyMedium,
+                                    ),
+                                    Flexible(
+                                      flex: 2,
+                                      child: Text(
+                                        description,
+                                        style: Theme.of(context)
+                                            .textTheme
+                                            .bodyMedium,
+                                      ),
+                                    ),
+                                  ],
                                 ),
                               ),
-                              itemBuilder: (BuildContext context) =>
-                                  <PopupMenuEntry>[
-                                PopupMenuItem(
-                                  onTap: () => onRsvp(context, ev, 'Yes'),
-                                  child: const Text('Yes'),
-                                ),
-                                PopupMenuItem(
-                                  onTap: () => onRsvp(context, ev, 'Maybe'),
-                                  child: const Text('Maybe'),
-                                ),
-                                PopupMenuItem(
-                                  onTap: () => onRsvp(context, ev, 'No'),
-                                  child: const Text('No'),
-                                ),
-                              ],
-                            ),
-                          ],
+                              const SizedBox(height: 15),
+                            ],
+                          ),
                         ),
                       ),
-                    ],
-                  ),
+                    ),
+                    const SizedBox(height: 15),
+                    ToggleButtons(
+                      isSelected: rsvp,
+                      onPressed: (index) async {
+                        var status = '';
+                        if (index == 0) status = 'No';
+                        if (index == 1) status = 'Maybe';
+                        if (index == 2) status = 'Yes';
+                        await onRsvp(context, event.value!, status);
```

Comment:
that, too, is pretty weird. it should at least be a else-if so they are skipped on the first match...